### ISSUE TITLE
Possible improvements to GNN

### ISSUE THREAD

**Issue by smsharma** [smsharma/set-diffuser#16]
- [x] The edge updates are not being propagated at the moment through the skip connections (only node updates are) -- should they?
- [x] Attention implementation (e.g., softmax not needed in attention _logits_ fn?).
